src/lowlevel/lowleveltransport.js

```diff
@@ -0,0 +1,286 @@
+/* @flow */
+
+import {patch} from './protobuf/monkey_patch';
+patch();
+
+import {create as createDefered} from '../defered';
+import {parseConfigure} from './protobuf/parse_protocol';
+import {buildAndSend} from './send';
+import {receiveAndParse} from './receive';
+import {resolveTimeoutPromise} from '../defered';
+
+// eslint-disable-next-line quotes
+const stringify = require('json-stable-stringify');
+
+import type {LowlevelTransportSharedPlugin, TrezorDeviceInfoDebug} from './sharedPlugin';
+import type {Defered} from '../defered';
+import type {Messages} from './protobuf/messages';
+import type {MessageFromTrezor, TrezorDeviceInfoWithSession, AcquireInput} from '../transport';
+
+import {debugInOut} from '../debug-decorator';
+
+function stableStringify(devices: ?Array<TrezorDeviceInfoWithSession>): string {
+  if (devices == null) {
+    return `null`;
+  }
+
+  const pureDevices = devices.map(device => {
+    const path = device.path;
+    const session = device.session == null ? null : device.session;
+    return {path, session};
+  });
+
+  return stringify(pureDevices);
+}
+
+function compare(a: TrezorDeviceInfoWithSession, b: TrezorDeviceInfoWithSession): number {
+  if (!isNaN(parseInt(a.path))) {
+    return parseInt(a.path) - parseInt(b.path);
```

Comment:
do we know the base? if yes, specify it
also, `!isNaN(parseInt(x))` is not a good check -- `parseInt('1a2b')` is `1`

Reply:
removed the numeric comparison since it isn't used anyways